UHDSoapyDevice.cpp

```diff
@@ -286,17 +286,31 @@ UHDSoapyDevice::~UHDSoapyDevice(void)
     SoapySDR::Device::unmake(_device);
 }
 
-void UHDSoapyDevice::setupChannelHooks(const int dir)
+void UHDSoapyDevice::setupChannelHooks()
 {
-    const std::string dirName((dir==SOAPY_SDR_RX)?"rx":"tx");
-    for (size_t ch = 0; ch < _device->getNumChannels(dir); ch++)
+    static const std::string kRxDirName = "rx";
+    static const std::string kTxDirName = "tx";
+    const size_t numRxChannels = _device->getNumChannels(SOAPY_SDR_RX);
+    const size_t numTxChannels = _device->getNumChannels(SOAPY_SDR_TX);
+
+    //We have to build up the same number of TX and RX channels to make UHD
+    //happy. If there are less channels in one direction than another, we fill
+    //in the direction with dummy channels.
+    const size_t numChannels = (numRxChannels >= numTxChannels) ? numRxChannels : numTxChannels;
```

Comment:
ok looks good, how-about `const size_t numChannels = std::max(numRxChannels, numTxChannels);` :-)

Reply:
Agreed. That makes things much more readable.